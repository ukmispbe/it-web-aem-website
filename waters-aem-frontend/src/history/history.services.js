import 'whatwg-fetch';
import { postDataRedirect } from '../utils/serviceFunctions';

class HistoryService {
    constructor(url) {
        this.url = url;
    }

    getOrderListPost(url, fromDate, poNumber, orderNumber, setError) {
        let options = {};
        options.orderNumber = orderNumber;
        options.purchaseOrderNumber = poNumber;
        options.fromDate = fromDate;
        options.maxRecs = "";

        return postDataRedirect(url, options, setError);
    }

    
    getQuoteListPost(url, fromDate, poNumber, orderNumber, setError) {
        let options = {};
        options.orderNumber = orderNumber;
        options.purchaseOrderNumber = poNumber;
        options.fromDate = fromDate;
        options.maxRecs = "";

        return postDataRedirect(url, options, setError);
    }
}

export default HistoryService;