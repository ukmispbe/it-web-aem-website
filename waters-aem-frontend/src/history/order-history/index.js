import React, { Component } from 'react';
import ReactPaginate from 'react-paginate';
import ReactSVG from 'react-svg';
import PropTypes from 'prop-types';
import HistoryService from'../history.services';
import OrderListItem from '../components/order-list-item';
import CountHeader from '../../common/count-header'
import TimePeriodDropdown from '../components/time-period-dropdown';
import FilterDropdown from '../components/filter-dropdown';
import Tabs from '../../navigation/tabs';
import Spinner from '../../utils/spinner';
import Analytics, { analyticTypes, setClickAnalytics, setSelectDropdownAnalytics } from '../../analytics';

import '../../styles/order-history.scss';
class OrderHistory extends Component {
    constructor(props) {
        super(props);
        const today = new Date();
        this.state = {
            listItems: "",
            fromDate: new Date(today.setDate(today.getDate() - 30)),
            poNumber: "",
            orderNumber: "",
            activeTabFilter: "All",
            activeIndex: 0,
            activeTimePeriod: 1,
            errorObjHistory: {},
            loading: true,
            noResults: false,
            error: false,
            initialPageLoad: true
        }

        this.page = {
            name: "Order History",
            type: "Orders",
            analytics: {
                reference: "orderHistory",
                timePeriod: "Order Period Selected",
                timePeriodOptions: ['Last 30 Days', 'Last 6 Months', 'Last 12 Months', 'Show All']
            }
        }

        this.paginationDefaults = {
            visibleRows: 10,
            nextIcon: "/content/dam/waters/en/brand-assets/icons/right.svg",
            previousIcon: "/content/dam/waters/en/brand-assets/icons/left.svg",
            pageRangeDisplayed: 8,
            marginPagesDisplayed: 1
        }
    }

    componentDidMount() {
        const {fromDate, poNumber, orderNumber, activeTabFilter} = this.state;
        this.retrieveData(fromDate, poNumber, orderNumber, activeTabFilter);
    }

    setAnalytics = (event, detail={}) => {
        const model = {
            detail,
            event
        };
        Analytics.setAnalytics(analyticTypes[this.page.analytics.reference].name, model);
    }

    setError = (error) => {
        this.setAnalytics('error', {error});
        this.setState({error: true})
    }

    handleCategorySelected(e) {
        // 0 = All Orders, 1 = Open Orders
        let tabId;
        let activeTabFilter = "All";
        (e.value || e.value === 0) ? tabId = e.value : tabId = e;

        if (tabId === 1) {
            activeTabFilter = "Open";
        }

        setClickAnalytics(this.page.title, `${this.page.title} ${activeTabFilter} ${this.page.type}`, '#');

        this.setState({
            activeTabFilter: activeTabFilter,
            activeIndex: tabId
        }, () => {
            const {fromDate, poNumber, orderNumber, activeTabFilter} = this.state 
            this.retrieveData(fromDate, poNumber, orderNumber, activeTabFilter);
        }); 
    }

    timePeriodHandler(e) {
        const { timePeriod, timePeriodOptions } = this.page.analytics;
        const selectedTimeframe = e.value;
        let now = new Date();
        let timeValue ='';
        const days = 30;
        const sixMonths = 6;
        const twelveMonths = 12;
        const allTime = 15;

        setSelectDropdownAnalytics(timePeriod, `${this.page.title} ${timePeriodOptions[selectedTimeframe]}`);

        switch (selectedTimeframe) {
            case 1:
                timeValue = new Date(now.setDate(now.getDate() - days));
                break;

            case 2:
                timeValue = new Date(now.setMonth(now.getMonth() - sixMonths));
                break;

            case 3:
                timeValue = new Date(now.setMonth(now.getMonth() - twelveMonths));
                break;

            case 4:
                timeValue = new Date(now.setMonth(now.getMonth() - allTime));
                break;
            default:
        }

        this.setState({
            fromDate: timeValue.toISOString(),
            activeTimePeriod: selectedTimeframe
        },() => {
            const {fromDate, poNumber, orderNumber, activeTabFilter} = this.state 
            this.retrieveData(fromDate, poNumber, orderNumber, activeTabFilter);
        });
    }

    setNoResultsState = () => {
        this.setState({ 
            listItems: null,
            pageCount: 0,
            listCount: 0,
            currentPage: 1,
            noResults: true,
            loading: false
        }); 
    }

    setResultsState = (filteredListItems) => {
        this.setState({ 
            listItems: filteredListItems,
            pageCount: Math.ceil(filteredListItems.length / this.paginationDefaults.visibleRows),
            listCount: filteredListItems.length,
            currentPage: 1,
            noResults: false,
            loading: false
        });
    }

    retrieveData = async (fromDate, poNumber, orderNumber, activeTabFilter) => {
        const HistoryServiceObj = new HistoryService();
        const fetchEndPoint = this.props.configs.fetchEndPoint;
        const orders = await HistoryServiceObj.getOrderListPost(fetchEndPoint, fromDate, poNumber, orderNumber, this.setError);

        if(orders && orders.length > 0){
            let filteredListItems = orders;
            if (activeTabFilter !== undefined && activeTabFilter !== "All" && activeTabFilter === "Open"){
                filteredListItems = orders.filter(function(i) {
                    return i.deliveryStatus === "Open" || i.deliveryStatus === "Partial";
                })
                if (filteredListItems.length > 0){
                    this.setResultsState(filteredListItems)
                } else {
                    this.setNoResultsState()
                }
            } else {
                this.setResultsState(filteredListItems)
            }
        } else {
            this.setNoResultsState()
        }

        !this.state.error && this.state.initialPageLoad && this.setAnalytics('load');
        this.setState({initialPageLoad: false});
    }

    renderTabs = () => {
        return (                     
            <Tabs className="cmp-search__categories-tabs"
                items={this.props.configs.tabs}
                activeIndex={this.state.activeIndex}
                onClick={e => this.handleCategorySelected(e)}
                enableFading={true}
            />
        );
    }

    renderDropDowns = () => {
        return (
            <div className="cmp-order-list__dropdowns">
                <FilterDropdown
                    onChange={e => this.handleCategorySelected(e)}
                    dropdownfilters={this.props.configs.dropdownfilters}
                />
                <TimePeriodDropdown 
                    onChange={e => this.timePeriodHandler(e)}
                    timePeriod={this.props.configs.timeperiod}
                />
            </div>
        );
    }

    renderCountHeader = () => {
        return (
            <CountHeader
                rows={this.paginationDefaults.visibleRows}
                count={this.state.listCount}
                current={this.state.currentPage}
                resultsText={this.props.configs.resultsText}
                noResultsText={this.props.configs.noResultsFoundTitle}
            />
        );
    }

    renderPaginatedResults = () => {
        const rows = this.paginationDefaults.visibleRows;
        const count = this.state.listCount;
        const current = this.state.currentPage;
        const endResults = count > current * rows ? current * rows : count;
        const startResults = current * rows - rows;
        let itemsToRender = this.state.listItems.slice(startResults, endResults);
        return itemsToRender;
    }

    paginationClickHandler = (page) => {
        this.setState({ 
            currentPage: page.selected + 1
        }); 
        window.scroll(0,0);
    }

    renderPagination() {
        if (this.state.listCount > this.paginationDefaults.visibleRows) {
            const previousIcon = (
                <ReactSVG src={this.paginationDefaults.previousIcon} />
            );
            const nextIcon = (
                <ReactSVG src={this.paginationDefaults.nextIcon} />
            );

            return (
                <ReactPaginate
                    pageCount={this.state.pageCount}
                    forcePage={this.state.currentPage -1}
                    pageRangeDisplayed={this.paginationDefaults.pageRangeDisplayed}
                    marginPagesDisplayed={this.paginationDefaults.marginPagesDisplayed}
                    containerClassName="paginate__container"
                    onPageChange={page => this.paginationClickHandler(page)}
                    breakLabel={'â€¦'}
                    previousLabel={previousIcon}
                    nextLabel={nextIcon}
                    initialPage={this.state.currentPage -1}
                    disableInitialCallback={true}
                    hrefBuilder={this.buildHref}
                />
            )
        } else {
            return <></>;
        }
    }

    renderNoResults = () => {
        return (
            <>
                <div className="cmp-order-list__no-results">
                    <p data-locator="no-results">{this.props.configs.noResultsFoundText}</p>
                    <p><a href={this.props.configs.shopAllHref} data-locator="shop-all">{this.props.configs.shopAllTitle}</a></p>
                </div>
            </>
        );
    }

    render() {
        return (
            <>  
                {this.state.loading ? ( <Spinner loading={this.state.loading} /> ) : null}
                {!this.state.loading && (
                    <>   
                    {this.renderTabs()}
                        <div className="cmp-order-list__header clearfix" data-locator="order-list-header-clearfix">
                            {this.renderDropDowns()}
                            {this.renderCountHeader()}
                        </div>

                        {this.state.noResults && this.renderNoResults()}

                        {this.state.listCount > 0 && this.renderPaginatedResults().map((item, index) => (               
                            <OrderListItem
                                data={item}
                                numberText={this.props.configs.numberText}
                                itemsText={this.props.configs.itemsText}
                                shipment={this.props.configs.shipment}
                                icons={this.props.configs.icons}
                            />
                        ))}
                        {this.state.listCount > 0 && this.renderPagination()}           
                    </>
                )}
            </>
        );
    }
}

OrderHistory.propTypes = {
    configs: PropTypes.object.isRequired
};

export default OrderHistory;