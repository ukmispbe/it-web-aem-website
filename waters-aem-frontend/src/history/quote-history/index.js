import React, { Component } from 'react';
import ReactPaginate from 'react-paginate';
import ReactSVG from 'react-svg';
import PropTypes from 'prop-types';
import HistoryService from'../history.services';
import QuoteListItem from '../components/quote-list-item';
import CountHeader from '../../common/count-header'
import TimePeriodDropdown from '../components/time-period-dropdown';
import FilterDropdown from '../components/filter-dropdown';
import Tabs from '../../navigation/tabs';
import Spinner from '../../utils/spinner';
import {getSoldToId, getDummySoldToId, getUserId} from '../../utils/userFunctions'
import Analytics, { analyticTypes, setClickAnalytics, setSelectDropdownAnalytics } from '../../analytics';

class QuoteHistory extends Component {
    constructor(props) {
        super(props);
        this.state = {
            listItems: "",
            noOfMonths: 1,
            poNumber: "",
            orderNumber: "",
            activeTabFilter: "ALL",
            activeIndex: 0,
            activeTimePeriod: 1,
            errorObjHistory: {},
            loading: true,
            noResults: false,
            error: false,
            initialPageLoad: true,
            currentPage:0,
            pageSize:10
        }

        this.page = {
            name: "Quote History",
            type: "Quotes",
            analytics: {
                reference: "quoteHistory",
                timePeriod: "Quote Period Selected",
                timePeriodOptions: ['Last 30 Days', 'Last 6 Months', 'Last 12 Months', 'Show All']
            }
        }

        this.paginationDefaults = {
            visibleRows: 10,
            nextIcon: "/content/dam/waters/en/brand-assets/icons/right.svg",
            previousIcon: "/content/dam/waters/en/brand-assets/icons/left.svg",
            pageRangeDisplayed: 8,
            marginPagesDisplayed: 1
        }
    }

    componentDidMount() {
        const {noOfMonths, activeTabFilter} = this.state;
        this.retrieveData(noOfMonths, activeTabFilter);
    }

    setAnalytics = (event, detail={}) => {
        const model = {
            detail,
            event
        };
        Analytics.setAnalytics(analyticTypes[this.page.analytics.reference].name, model);
    }

    setError = (error) => {
        this.setAnalytics('error', {error});
        this.setState({error: true})
    }

    handleCategorySelected(e) {
        // 0 = All Quotes, 1 = Open Quotes, 2 = Closed Quotes
        let tabId;
        let activeTabFilterStatus = "ALL";
        (e.value || e.value === 0) ? tabId = e.value : tabId = e;

        if (tabId === 1) {
            activeTabFilterStatus = "OPEN";
        } else if (tabId === 2) {
            activeTabFilterStatus = "CLOSED";
        }

        setClickAnalytics(this.page.title, `${this.page.title} ${activeTabFilterStatus} ${this.page.type}`, '#');

        this.setState({
            activeTabFilter: activeTabFilterStatus,
            activeIndex: tabId,
            currentPage:0
        }, () => {
            const {noOfMonths, activeTabFilter} = this.state 
            this.retrieveData(noOfMonths, activeTabFilter);
        }); 
    }

    timePeriodHandler(e) {
        const { timePeriod, timePeriodOptions } = this.page.analytics;
        const selectedTimeframe = e.value;
        let timeValue ='';
        const month = 1;
        const sixMonths = 6;
        const twelveMonths = 12;
        const allTime = 0;

        setSelectDropdownAnalytics(timePeriod, `${this.page.title} ${timePeriodOptions[selectedTimeframe -1]}`);

        switch (selectedTimeframe) {
            case 1:
                timeValue = month;
                break;

            case 2:
                timeValue = sixMonths;
                break;

            case 3:
                timeValue = twelveMonths;
                break;

            case 4:
                timeValue = allTime;
                break;
            default:
        }

        this.setState({
            noOfMonths: timeValue,
            activeTimePeriod: selectedTimeframe,
            currentPage:0
        },() => {
            const {noOfMonths, activeTabFilter} = this.state 
            this.retrieveData(noOfMonths, activeTabFilter);
        });
    }

    setNoResultsState = () => {
        this.setState({ 
            listItems: null,
            pageCount: 0,
            listCount: 0,
            currentPage: 0,
            noResults: true,
            loading: false,
        }); 
    }

    setResultsState = (quoteData) => {
        const {totalNumberOfResults = 0, quotes=[], numberOfPages, currentPage} = quoteData || {}
        this.setState({ 
            listItems: quotes,
            pageCount: numberOfPages,
            listCount: totalNumberOfResults,
            currentPage: currentPage,
            noResults: false,
            loading: false,
        });
    }

    getQueryParam = (fetchEndPoint,noOfMonths,activeTabFilter) =>{
      let url = fetchEndPoint;
      const {currentPage, pageSize} = this.state;
      const  userId = getUserId();
      const soldToId = getSoldToId() || getDummySoldToId();
      let queryParam = `userId=${userId}&soldToId=${soldToId}&currentPage=${currentPage}&pageSize=${pageSize}&fields=FULL`;
      if(activeTabFilter && activeTabFilter !== "ALL"){
        queryParam = `${queryParam}&state=${activeTabFilter}`;
      }
      if(noOfMonths){
        queryParam = `${queryParam}&duration=${noOfMonths}`;
      }
      return `${url}?${queryParam}`;
    }

    retrieveData = async (noOfMonths, activeTabFilter) => {
        const HistoryServiceObj = new HistoryService();
        let fetchEndPoint = this.props.configs.fetchEndPoint;
        fetchEndPoint = this.getQueryParam(fetchEndPoint,noOfMonths,activeTabFilter)
        const quoteData = await HistoryServiceObj.getQuoteHistory(fetchEndPoint);
        const {totalNumberOfResults = 0} = quoteData
        if(quoteData && totalNumberOfResults > 0){
            this.setResultsState(quoteData)
        } else {
            this.setNoResultsState()
        }

        !this.state.error && this.state.initialPageLoad && this.setAnalytics('load');
        this.setState({initialPageLoad: false});
    }

    renderTabs = () => {
        const {tabs = []} = this.props.configs || {};
        return (                     
            <Tabs className="cmp-search__categories-tabs"
                items={tabs}
                activeIndex={this.state.activeIndex}
                onClick={e => this.handleCategorySelected(e)}
                enableFading={true}
            />
        );
    }

    renderDropDowns = () => {
        return (
            <div className="cmp-order-list__dropdowns">
                <FilterDropdown
                    onChange={e => this.handleCategorySelected(e)}
                    dropdownfilters={this.props.configs.dropdownfilters}
                />
                <TimePeriodDropdown 
                    onChange={e => this.timePeriodHandler(e)}
                    timePeriod={this.props.configs.timeperiod}
                />
            </div>
        );
    }

    renderCountHeader = () => {
        return (
            <CountHeader
                rows={this.state.pageSize}
                count={this.state.listCount}
                current={this.state.currentPage + 1}
                resultsText={this.props.configs.resultsText}
                noResultsText={this.props.configs.noResultsFoundTitle}
            />
        );
    }

    paginationClickHandler = (page) => {
        this.setState({ 
            currentPage: page.selected
        },() => {
            const {noOfMonths, activeTabFilter} = this.state 
            this.retrieveData(noOfMonths, activeTabFilter);
        }); 
        window.scroll(0,0);
    }

    renderPagination() {
        if (this.state.listCount > this.state.pageSize) {
            const previousIcon = (
                <ReactSVG src={this.paginationDefaults.previousIcon} />
            );
            const nextIcon = (
                <ReactSVG src={this.paginationDefaults.nextIcon} />
            );

            return (
                <ReactPaginate
                    pageCount={this.state.pageCount}
                    forcePage={this.state.currentPage}
                    pageRangeDisplayed={this.paginationDefaults.pageRangeDisplayed}
                    marginPagesDisplayed={this.paginationDefaults.marginPagesDisplayed}
                    containerClassName="paginate__container"
                    onPageChange={page => this.paginationClickHandler(page)}
                    breakLabel={'â€¦'}
                    previousLabel={previousIcon}
                    nextLabel={nextIcon}
                    initialPage={this.state.currentPage}
                    disableInitialCallback={true}
                    hrefBuilder={this.buildHref}
                />
            )
        } else {
            return <></>;
        }
    }

    renderNoResults = () => {
        return (
            <>
                <div className="cmp-order-list__no-results">
                    <p data-locator="no-results">{this.props.configs.noResultsFoundText}</p>
                    <p><a href={this.props.configs.shopAllHref} data-locator="shop-all">{this.props.configs.shopAllTitle}</a></p>
                </div>
            </>
        );
    }

    render() {
        const {listCount,listItems, noResults, loading} = this.state;
        return (
            <>  
                {loading ? ( <Spinner loading={loading} /> ) : null}
                {!loading && (
                    <>   
                    {this.renderTabs()}
                        <div className="cmp-order-list__header clearfix" data-locator="order-list-header-clearfix">
                            {this.renderDropDowns()}
                            {this.renderCountHeader()}
                        </div>

                        {noResults && this.renderNoResults()}

                        {listCount > 0 && listItems.map((item, index) => (               
                            <QuoteListItem
                                data={item}
                                numberText={this.props.configs.numberText}
                                itemsText={this.props.configs.itemsText}
                                shipment={this.props.configs.shipment}
                                created={this.props.configs.created}
                                expires={this.props.configs.expires}
                                orderNumberText={this.props.configs.orderNumberText}
                                icons={this.props.configs.icons}
                                quoteAgainTitle={this.props.configs.quoteAgainTitle}
                                index={index}
                                isShowQuoteAgainButton={this.props.configs.isShowQuoteAgainButton}
								newQuote={this.props.configs.newQuote}
                            />
                        ))}
                        {listCount > 0 && this.renderPagination()}           
                    </>
                )}
            </>
        );
    }
}

QuoteHistory.propTypes = {
    configs: PropTypes.object.isRequired
};

export default QuoteHistory;