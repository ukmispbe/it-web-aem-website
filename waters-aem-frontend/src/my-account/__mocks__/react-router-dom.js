import React from 'react';
import * as rrd from "react-router-dom";

// We need to mock it because it will conflict with the MemoryRouter and the initialEntries will not work
rrd.HashRouter = jest.fn(({children}) => <div>{children}</div>);
module.exports = rrd;