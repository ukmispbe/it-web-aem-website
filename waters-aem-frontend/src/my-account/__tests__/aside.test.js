import React from "react";
import renderer from "react-test-renderer";
import { createBrowserHistory } from 'history';
import { HashRouter } from 'react-router-dom';
import Aside from "../aside";
import props from "../__mocks__/en_US/my-account-json";

const history = createBrowserHistory();

describe("Feature: Aside Component for my account", () => {
    beforeAll(() => {
        console.warn = jest.fn();
    });

    afterEach(() => {
        jest.restoreAllMocks();
    });

    describe("Scenario: Rendering", () => {
        describe("When no link is active", () => {
            it("Then it should match snapshot", () => {
                history.push("http://localhost.com/my-account.html");

                const mockProps = {
                    ...props,
                    history
                };

                const json = renderer.create(<HashRouter><Aside {...mockProps} /></HashRouter>);

                expect(json).toMatchSnapshot();
            });
        });

        describe("When profile link is active", () => {
            it("Then it should match snapshot", () => {
                history.push("http://localhost.com/my-account.html#profile");

                const mockProps = {
                    ...props,
                    history
                };

                const json = renderer.create(<HashRouter><Aside {...mockProps} /></HashRouter>);

                expect(json).toMatchSnapshot();
            });
        });

        describe("When password link is active", () => {
            it("Then it should match snapshot", () => {
                history.push("http://localhost.com/my-account.html#changepassword");

                const mockProps = {
                    ...props,
                    history
                };

                const json = renderer.create(<HashRouter><Aside {...mockProps} /></HashRouter>);

                expect(json).toMatchSnapshot();
            });
        });

        describe("When orders link is active", () => {
            it("Then it should match snapshot", () => {
                history.push("http://localhost.com/my-account.html#orderhistory");

                const mockProps = {
                    ...props,
                    history
                };

                const json = renderer.create(<HashRouter><Aside {...mockProps} /></HashRouter>);

                expect(json).toMatchSnapshot();
            });
        });
    });
});