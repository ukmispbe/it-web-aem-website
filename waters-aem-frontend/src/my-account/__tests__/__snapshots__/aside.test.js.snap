// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: Aside Component for my account Scenario: Rendering When ecommerce is disabled Then it should match snapshot 1`] = `
<div
  className="cmp-my-account__aside-wrapper"
>
  <div
    className="cmp-title"
  >
    <h1
      className="cmp-title__text"
    >
      Profile
    </h1>
    <div
      className="cmp-title__print"
    >
      Profile
    </div>
    <div
      className="cmp-accent-rule"
    >
      <hr />
    </div>
  </div>
  <div
    className="cmp-my-account__aside-links"
  >
    <div
      className="tile"
    >
      <div
        className="tile__title"
      >
        Account Information
      </div>
      <div
        className="tile__links"
      >
        <span
          className="link--active"
        >
          Profile
        </span>
        <a
          href="/changepassword"
          onClick={[Function]}
        >
          Password
        </a>
      </div>
    </div>
    <div
      className="tile"
    >
      <div
        className="tile__title"
      >
        Additional Links
      </div>
      <div
        className="tile__links"
      >
        <a
          href="https://www.waters.com/waters/nav.htm?cid=10006027"
        >
          Request Technical Support
        </a>
        <a
          href="https://www.waters.com/waters/nav.htm?cid=134614448"
        >
          Waters Knowledgebase
        </a>
        <a
          href="https://www.waters.com/waters/nav.htm?cid=513247"
        >
          Training Courses
        </a>
        <a
          href="https://www.waters.com/waters/localeRedirect.htm?type=contact_us"
        >
          Contact Waters
        </a>
      </div>
    </div>
  </div>
  <div
    className="cmp-my-account__aside-content"
  />
</div>
`;

exports[`Feature: Aside Component for my account Scenario: Rendering When no link is active Then it should match snapshot 1`] = `
<div
  className="cmp-my-account__aside-wrapper"
>
  <div
    className="cmp-title"
  >
    <h1
      className="cmp-title__text"
    >
      
    </h1>
    <div
      className="cmp-title__print"
    >
      
    </div>
    <div
      className="cmp-accent-rule"
    >
      <hr />
    </div>
  </div>
  <div
    className="cmp-my-account__aside-links"
  >
    <div
      className="tile"
    >
      <div
        className="tile__title"
      >
        Account Information
      </div>
      <div
        className="tile__links"
      >
        <a
          href="/profile"
          onClick={[Function]}
        >
          Profile
        </a>
        <a
          href="/changepassword"
          onClick={[Function]}
        >
          Password
        </a>
      </div>
    </div>
    <div
      className="tile"
    >
      <div
        className="tile__title"
      >
        Orders
      </div>
      <div
        className="tile__links"
      >
        <a
          href="/orderhistory"
          onClick={[Function]}
        >
          Order History
        </a>
      </div>
    </div>
    <div
      className="tile"
    >
      <div
        className="tile__title"
      >
        Additional Links
      </div>
      <div
        className="tile__links"
      >
        <a
          href="https://www.waters.com/waters/nav.htm?cid=10006027"
        >
          Request Technical Support
        </a>
        <a
          href="https://www.waters.com/waters/nav.htm?cid=134614448"
        >
          Waters Knowledgebase
        </a>
        <a
          href="https://www.waters.com/waters/nav.htm?cid=513247"
        >
          Training Courses
        </a>
        <a
          href="https://www.waters.com/waters/localeRedirect.htm?type=contact_us"
        >
          Contact Waters
        </a>
      </div>
    </div>
  </div>
  <div
    className="cmp-my-account__aside-content"
  />
</div>
`;

exports[`Feature: Aside Component for my account Scenario: Rendering When password link is active Then it should match snapshot 1`] = `
<div
  className="cmp-my-account__aside-wrapper"
>
  <div
    className="cmp-title"
  >
    <h1
      className="cmp-title__text"
    >
      Password
    </h1>
    <div
      className="cmp-title__print"
    >
      Password
    </div>
    <div
      className="cmp-accent-rule"
    >
      <hr />
    </div>
  </div>
  <div
    className="cmp-my-account__aside-links"
  >
    <div
      className="tile"
    >
      <div
        className="tile__title"
      >
        Account Information
      </div>
      <div
        className="tile__links"
      >
        <a
          href="/profile"
          onClick={[Function]}
        >
          Profile
        </a>
        <span
          className="link--active"
        >
          Password
        </span>
      </div>
    </div>
    <div
      className="tile"
    >
      <div
        className="tile__title"
      >
        Orders
      </div>
      <div
        className="tile__links"
      >
        <a
          href="/orderhistory"
          onClick={[Function]}
        >
          Order History
        </a>
      </div>
    </div>
    <div
      className="tile"
    >
      <div
        className="tile__title"
      >
        Additional Links
      </div>
      <div
        className="tile__links"
      >
        <a
          href="https://www.waters.com/waters/nav.htm?cid=10006027"
        >
          Request Technical Support
        </a>
        <a
          href="https://www.waters.com/waters/nav.htm?cid=134614448"
        >
          Waters Knowledgebase
        </a>
        <a
          href="https://www.waters.com/waters/nav.htm?cid=513247"
        >
          Training Courses
        </a>
        <a
          href="https://www.waters.com/waters/localeRedirect.htm?type=contact_us"
        >
          Contact Waters
        </a>
      </div>
    </div>
  </div>
  <div
    className="cmp-my-account__aside-content"
  />
</div>
`;

exports[`Feature: Aside Component for my account Scenario: Rendering When profile link is active Then it should match snapshot 1`] = `
<div
  className="cmp-my-account__aside-wrapper"
>
  <div
    className="cmp-title"
  >
    <h1
      className="cmp-title__text"
    >
      Profile
    </h1>
    <div
      className="cmp-title__print"
    >
      Profile
    </div>
    <div
      className="cmp-accent-rule"
    >
      <hr />
    </div>
  </div>
  <div
    className="cmp-my-account__aside-links"
  >
    <div
      className="tile"
    >
      <div
        className="tile__title"
      >
        Account Information
      </div>
      <div
        className="tile__links"
      >
        <span
          className="link--active"
        >
          Profile
        </span>
        <a
          href="/changepassword"
          onClick={[Function]}
        >
          Password
        </a>
      </div>
    </div>
    <div
      className="tile"
    >
      <div
        className="tile__title"
      >
        Orders
      </div>
      <div
        className="tile__links"
      >
        <a
          href="/orderhistory"
          onClick={[Function]}
        >
          Order History
        </a>
      </div>
    </div>
    <div
      className="tile"
    >
      <div
        className="tile__title"
      >
        Additional Links
      </div>
      <div
        className="tile__links"
      >
        <a
          href="https://www.waters.com/waters/nav.htm?cid=10006027"
        >
          Request Technical Support
        </a>
        <a
          href="https://www.waters.com/waters/nav.htm?cid=134614448"
        >
          Waters Knowledgebase
        </a>
        <a
          href="https://www.waters.com/waters/nav.htm?cid=513247"
        >
          Training Courses
        </a>
        <a
          href="https://www.waters.com/waters/localeRedirect.htm?type=contact_us"
        >
          Contact Waters
        </a>
      </div>
    </div>
  </div>
  <div
    className="cmp-my-account__aside-content"
  />
</div>
`;
